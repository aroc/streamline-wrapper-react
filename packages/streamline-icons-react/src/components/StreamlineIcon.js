import React from 'react';
import PropTypes from 'prop-types';
import humps from 'humps';

import './StreamlineIcon.css';

const StreamlineIcon = ({icon, fill, size, stroke, width, height, customClassName, spin, pulse, infinite, fast, easeInOut}) => {
  console.debug('In custom StreamlineIcon')

  const getClassName = () => {

    let className = [`Streamline_Icon`];
    if (spin) className.push('Streamline_Icon_Spin', 'Streamline_Icon_Animated');
    if (pulse) className.push('Streamline_Icon_Pulse', 'Streamline_Icon_Animated');
    if (infinite) className.push('Streamline_Animation_Infinite');
    if (fast) className.push('Streamline_Animation_Fast');
    if (easeInOut) className.push('Streamline_Animation_EaseInOut');
    if (customClassName) className.push(customClassName);

    return [...new Set(className)].join(' ');

  };

  const getSize = () => {

    //Default
    const sizeObject = {
      width: icon[1],
      height: icon[2],
      isDefault: true,
    };

    if(size) {

      sizeObject.isDefault = size === sizeObject.width;
      sizeObject.height = size;
      sizeObject.width = size;

    } else {

      if(height && parseInt(height) !== sizeObject.height) {
        sizeObject.height = height;
        sizeObject.isDefault = false;
      }

      if(width && parseInt(width) !== sizeObject.width) {
        sizeObject.width = width;
        sizeObject.isDefault = false;
      }

    }

    return sizeObject;

  };

  const getStyle = () => {

    const sizeObj = getSize();

    delete sizeObj.isDefault;

    return {...sizeObj};

  };

  const renderIcon = () => {

    const size = getSize();

    const renderPaths = () => {
      return icon[4].map(
        (path, index) => {

          const options = {...humps.camelizeKeys(icon[3][index])};

          options['stroke'] = stroke || icon[3][index]['stroke'];
          options['fill'] = fill || icon[3][index]['fill'];


          return <path
            {...options}
            key={Math.random().toString()}
            d={path}/>

        })
    };

    return (
      <span className={getClassName()}>
          <svg
            viewBox={`0 0 ${size.width} ${size.height}`}
            style={{...getStyle()}}
            width={size.width}
            height={size.height}
          >

            {!size.isDefault ?
              <g transform={`scale(${size.width / icon[1]},${size.height / icon[2]})`}>
                {renderPaths()}
              </g>
              :
              renderPaths()

            }

          </svg>
        </span>
    )
  };

  return  renderIcon();

};

StreamlineIcon.propTypes = {

  fill: PropTypes.string,
  stroke: PropTypes.string,

  icon: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.array])),

  height: PropTypes.number,
  width: PropTypes.number,
  size: PropTypes.number,

  pulse: PropTypes.bool,
  spin: PropTypes.bool,
  fast: PropTypes.bool,
  easeInOut: PropTypes.bool,
  infinite: PropTypes.bool,

  customClassName: PropTypes.string

};

StreamlineIcon.defaultProps = {

  size: 24,
  pulse: false,
  spin: false,
  fast: false,
  easeInOut: false,
  infinite: false,
  customClassName: null

};

export default StreamlineIcon
