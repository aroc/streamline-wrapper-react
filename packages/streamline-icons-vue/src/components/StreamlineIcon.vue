<template>
  <span
    class="Streamline_Icon"
    :class="{
      'Streamline_Icon_Spin': spin,
      'Streamline_Icon_Pulse': pulse,
      'Streamline_Animation_Infinite': infinite,
      'Streamline_Icon_Fast': fast,
      'Streamline_Icon_EaseInOut': easeInOut,
      'Streamline_Icon_Animated': spin || pulse
    }"
  >
    <svg
      :viewBox="`0 0 ${computedSize.width} ${computedSize.height}`"
      :style="`width: ${computedSize.width}; height: ${computedSize.height}`"
      :width="computedSize.width"
      :height="computedSize.height"
    >
      <g
        v-if="!computedSize.isDefault"
        :transform="`scale(${computedSize.width / icon[1]},${computedSize.height / icon[2]})`"
      >
        <icon-path
          v-for="(path, index) in icon[4]"
          :icon="icon"
          :path="path"
          :index="index"
          :stroke="stroke"
          :fill="fill"
          :key="index"
        />
      </g>
      <svg v-else>
          <icon-path
            v-for="(path, index) in icon[4]"
            :icon="icon"
            :path="path"
            :index="index"
            :stroke="stroke"
            :fill="fill"
            :key="index"
          />
      </svg>
    </svg>
  </span>
</template>

<script>
import IconPath from './IconPath'

export default {
  components: {
    IconPath
  },
  props: {
    icon: {
      type: [String, Number, Array, Object]
    },
    fill: {
      type: String
    },
    stroke: {
      type: String
    },
    size: {
      type: Number,
      default: 24
    },
    width: {
      type: Number
    },
    height: {
      type: Number
    },
    spin: {
      type: Boolean,
      default: false
    },
    pulse: {
      type: Boolean,
      default: false
    },
    infinite: {
      type: Boolean,
      default: false
    },
    fast: {
      type: Boolean,
      default: false
    },
    easeInOut: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    computedSize () {
      const { icon, size } = this
      const sizeObject = {
        width: icon[1],
        height: icon[2],
        isDefault: true
      }

      if (size) {
        sizeObject.isDefault = (size === sizeObject.width)
        sizeObject.height = size
        sizeObject.isDefault = size
      } else {
        if (this.height && parseInt(this.height) !== sizeObject.height) {
          sizeObject.height = this.height
          sizeObject.isDefault = false
        }

        if (this.width && parseInt(this.width) !== sizeObject.width) {
          sizeObject.width = this.width
          sizeObject.isDefault = false
        }
      }

      return sizeObject
    }
  }
}
</script>

<style>
.Streamline_Icon {
    display: flex;
    align-items: center;
    justify-content: center;
    box-sizing: content-box;
}

.Streamline_Icon_Animated {
    animation-duration: 1s;
    animation-fill-mode: both;
    transform-origin: center;
    animation-timing-function: linear;

}

.Streamline_Icon_Spin {
    animation-name: rotate;
    animation-duration: 2s;
}

.Streamline_Animation_Infinite {
    animation-iteration-count: infinite;
}

.Streamline_Animation_Fast {
    animation-duration: 1s;
}

.Streamline_Animation_EaseInOut {
    animation-timing-function: ease-in-out;
}

@keyframes rotate {
    100% {
        transform: rotate(360deg);
    }
}
</style>
