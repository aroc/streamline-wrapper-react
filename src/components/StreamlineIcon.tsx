import React, {FunctionComponent} from 'react';
import humps from 'humps';

import './StreamlineIcon.css';

type IconOptions = {
  fill: string
  stroke: string
  'stroke-linecap': string
  'stroke-linejoin': string
  'stroke-width': number
}
type Icon = [string, number, number, [IconOptions], [string]]

const StreamlineIcon: FunctionComponent<{
  icon: Icon
  fill?: string
  size: number
  stroke?: string
  width?: number
  height?: number
  customClassName?: string
  spin: boolean
  pulse: boolean
  infinite: boolean
  easeInOut: boolean
  fast: boolean
}> = ({
                              icon,
                              fill,
                              size= 24,
                              stroke,
                              width,
                              height,
                              customClassName,
                                spin= false,
                              pulse= false,
                                infinite= false,
                                fast= false,
                                easeInOut= false
}) => {
  const getClassName = () => {

    let className = [`Streamline_Icon`];
    if (spin) className.push('Streamline_Icon_Spin', 'Streamline_Icon_Animated');
    if (pulse) className.push('Streamline_Icon_Pulse', 'Streamline_Icon_Animated');
    if (infinite) className.push('Streamline_Animation_Infinite');
    if (fast) className.push('Streamline_Animation_Fast');
    if (easeInOut) className.push('Streamline_Animation_EaseInOut');
    if (customClassName) className.push(customClassName);

    return className.join(' ');

  };

  const getSize = () => {

    //Default
    const sizeObject = {
      width: icon[1],
      height: icon[2],
      isDefault: true,
    };

    if(size) {

      sizeObject.isDefault = size === sizeObject.width;
      sizeObject.height = size;
      sizeObject.width = size;

    } else {

      if(height && height !== sizeObject.height) {
        sizeObject.height = height;
        sizeObject.isDefault = false;
      }

      if(width && width !== sizeObject.width) {
        sizeObject.width = width;
        sizeObject.isDefault = false;
      }

    }

    return sizeObject;

  };

  const getStyle = () => {

    const sizeObj = getSize();

    delete sizeObj.isDefault;

    return {...sizeObj};

  };

  const renderIcon = () => {

    const size = getSize();

    const renderPaths = () => {
      return icon[4].map(
          (path, index) => {

            const options = {...humps.camelizeKeys(icon[3][index])};

            options['stroke'] = stroke || icon[3][index]['stroke'];
            options['fill'] = fill || icon[3][index]['fill'];


            return <path
                {...options}
                key={Math.random().toString()}
                d={path}/>

          })
    };

    return (
        <span className={getClassName()}>
          <svg
              viewBox={`0 0 ${size.width} ${size.height}`}
              style={{...getStyle()}}
              width={size.width}
              height={size.height}
          >

            {!size.isDefault ?
                <g transform={`scale(${size.width / icon[1]},${size.height / icon[2]})`}>
                  {renderPaths()}
                </g>
                :
                renderPaths()

            }

          </svg>
        </span>
    )
  };

  return  renderIcon();

};

export default StreamlineIcon
