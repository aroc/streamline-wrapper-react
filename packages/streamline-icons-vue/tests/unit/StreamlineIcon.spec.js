import { render } from '@vue/server-test-utils'
import { shallowMount, mount } from '@vue/test-utils'
import StreamlineIcon from '@/components/StreamlineIcon.vue'

const icons = {
  TouchIdLock: ['touch-id-lock', 24, 24, [{
    fill: 'none',
    stroke: '#000',
    'stroke-linecap': 'round',
    'stroke-linejoin': 'round',
    'stroke-width': '1.5'
  }, {
    fill: 'none',
    stroke: '#000',
    'stroke-linecap': 'round',
    'stroke-linejoin': 'round',
    'stroke-width': '1.5'
  }, {
    fill: 'none',
    stroke: '#000',
    'stroke-linecap': 'round',
    'stroke-linejoin': 'round',
    'stroke-width': '1.5'
  }, {
    fill: 'none',
    stroke: '#000',
    'stroke-linecap': 'round',
    'stroke-linejoin': 'round',
    'stroke-width': '1.5'
  }, {
    fill: 'none',
    stroke: '#000',
    'stroke-linecap': 'round',
    'stroke-linejoin': 'round',
    'stroke-width': '1.5'
  }, {
    fill: 'none',
    stroke: '#000',
    'stroke-linecap': 'round',
    'stroke-linejoin': 'round',
    'stroke-width': '1.5'
  }, {
    fill: 'none',
    stroke: '#000',
    'stroke-linecap': 'round',
    'stroke-linejoin': 'round',
    'stroke-width': '1.5'
  }, {
    fill: 'none',
    stroke: '#000',
    'stroke-linecap': 'round',
    'stroke-linejoin': 'round',
    'stroke-width': '1.5'
  }, {
    fill: 'none',
    stroke: '#000',
    'stroke-linecap': 'round',
    'stroke-linejoin': 'round',
    'stroke-width': '1.5'
  }, {
    fill: 'none',
    stroke: '#000',
    'stroke-linecap': 'round',
    'stroke-linejoin': 'round',
    'stroke-width': '1.5'
  }], ['M12,1.753A7.5,7.5,0,0,0,.75,8.25', 'M15.75,8.25a7.478,7.478,0,0,0-.624-3', 'M8.25,23.25a7.563,7.563,0,0,0,1.5-.15', 'M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876', 'M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352', 'M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244', 'M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3', 'M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z', 'M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z', 'M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375']]
}

describe('StreamlineIcon.vue', () => {
  it('renders without crashing', () => {
    const wrapper = shallowMount(StreamlineIcon, {
      propsData: {
        icon: icons.TouchIdLock
      }
    })
  })

  it('Testing Snapshots', () => {
    const wrapper = shallowMount(StreamlineIcon, {
      propsData: {
        icon: icons.TouchIdLock,
        stroke: '#000'
      }
    })

    expect(wrapper).toMatchSnapshot();

  });

  it('Contain SVG Node', () => {
    const wrapper = shallowMount(StreamlineIcon, {
      propsData: {
        icon: icons.TouchIdLock
      }
    })
    expect(wrapper.find('svg')).toBeTruthy();
  });

  it('Is colored correctly', () => {
    let shallow = new Shallow(StreamlineIcon)
    const color = '#ff6e61';
    const wrapper = mount(StreamlineIcon, {
      propsData: {
        icon: icons.TouchIdLock,
        stroke: color
      }
    })
    expect(wrapper.find('path').attributes('stroke')).toBe(color);

  })
})
