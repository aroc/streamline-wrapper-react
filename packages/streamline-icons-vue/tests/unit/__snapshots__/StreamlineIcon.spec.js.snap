// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StreamlineIcon.vue Testing Snapshots 1`] = `<span class="Streamline_Icon"><svg viewBox="0 0 24 24" width="24" height="24"><svg><icon-path-stub icon="touch-id-lock,24,24,[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],M12,1.753A7.5,7.5,0,0,0,.75,8.25,M15.75,8.25a7.478,7.478,0,0,0-.624-3,M8.25,23.25a7.563,7.563,0,0,0,1.5-.15,M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876,M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352,M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244,M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3,M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z,M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z,M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" index="0" path="M12,1.753A7.5,7.5,0,0,0,.75,8.25" stroke="#000"></icon-path-stub><icon-path-stub icon="touch-id-lock,24,24,[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],M12,1.753A7.5,7.5,0,0,0,.75,8.25,M15.75,8.25a7.478,7.478,0,0,0-.624-3,M8.25,23.25a7.563,7.563,0,0,0,1.5-.15,M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876,M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352,M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244,M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3,M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z,M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z,M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" index="1" path="M15.75,8.25a7.478,7.478,0,0,0-.624-3" stroke="#000"></icon-path-stub><icon-path-stub icon="touch-id-lock,24,24,[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],M12,1.753A7.5,7.5,0,0,0,.75,8.25,M15.75,8.25a7.478,7.478,0,0,0-.624-3,M8.25,23.25a7.563,7.563,0,0,0,1.5-.15,M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876,M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352,M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244,M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3,M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z,M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z,M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" index="2" path="M8.25,23.25a7.563,7.563,0,0,0,1.5-.15" stroke="#000"></icon-path-stub><icon-path-stub icon="touch-id-lock,24,24,[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],M12,1.753A7.5,7.5,0,0,0,.75,8.25,M15.75,8.25a7.478,7.478,0,0,0-.624-3,M8.25,23.25a7.563,7.563,0,0,0,1.5-.15,M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876,M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352,M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244,M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3,M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z,M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z,M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" index="3" path="M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876" stroke="#000"></icon-path-stub><icon-path-stub icon="touch-id-lock,24,24,[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],M12,1.753A7.5,7.5,0,0,0,.75,8.25,M15.75,8.25a7.478,7.478,0,0,0-.624-3,M8.25,23.25a7.563,7.563,0,0,0,1.5-.15,M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876,M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352,M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244,M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3,M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z,M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z,M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" index="4" path="M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352" stroke="#000"></icon-path-stub><icon-path-stub icon="touch-id-lock,24,24,[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],M12,1.753A7.5,7.5,0,0,0,.75,8.25,M15.75,8.25a7.478,7.478,0,0,0-.624-3,M8.25,23.25a7.563,7.563,0,0,0,1.5-.15,M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876,M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352,M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244,M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3,M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z,M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z,M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" index="5" path="M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244" stroke="#000"></icon-path-stub><icon-path-stub icon="touch-id-lock,24,24,[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],M12,1.753A7.5,7.5,0,0,0,.75,8.25,M15.75,8.25a7.478,7.478,0,0,0-.624-3,M8.25,23.25a7.563,7.563,0,0,0,1.5-.15,M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876,M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352,M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244,M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3,M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z,M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z,M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" index="6" path="M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3" stroke="#000"></icon-path-stub><icon-path-stub icon="touch-id-lock,24,24,[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],M12,1.753A7.5,7.5,0,0,0,.75,8.25,M15.75,8.25a7.478,7.478,0,0,0-.624-3,M8.25,23.25a7.563,7.563,0,0,0,1.5-.15,M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876,M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352,M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244,M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3,M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z,M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z,M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" index="7" path="M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z" stroke="#000"></icon-path-stub><icon-path-stub icon="touch-id-lock,24,24,[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],M12,1.753A7.5,7.5,0,0,0,.75,8.25,M15.75,8.25a7.478,7.478,0,0,0-.624-3,M8.25,23.25a7.563,7.563,0,0,0,1.5-.15,M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876,M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352,M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244,M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3,M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z,M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z,M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" index="8" path="M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z" stroke="#000"></icon-path-stub><icon-path-stub icon="touch-id-lock,24,24,[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],M12,1.753A7.5,7.5,0,0,0,.75,8.25,M15.75,8.25a7.478,7.478,0,0,0-.624-3,M8.25,23.25a7.563,7.563,0,0,0,1.5-.15,M.75,12.75v3a7.5,7.5,0,0,0,4.5,6.876,M12.75,12.75V8.25A4.5,4.5,0,0,0,6,4.352,M3.75,8.25v7.5a4.5,4.5,0,0,0,6,4.244,M6.75,15.75a1.5,1.5,0,0,0,3,0V8.25a1.5,1.5,0,0,0-3,0v3,M14.250 15.750 L23.250 15.750 L23.250 23.250 L14.250 23.250 Z,M18.75,11.25h0a3,3,0,0,0-3,3v1.5h6v-1.5A3,3,0,0,0,18.75,11.25Z,M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" index="9" path="M18.75,19.154a.375.375,0,1,1-.375.375.375.375,0,0,1,.375-.375" stroke="#000"></icon-path-stub></svg></svg></span>`;
